scripts/run_mcmc.py

```diff
@@ -0,0 +1,295 @@
+    --iteration 1000
+    or to read in a TreeNode
+    --init_tree_fp data/trees/tree_3.json
+    --init_TreeNode
+
+Config File .json:
+{
+    "move_probs": {
+        "prune_and_reattach": 0.1,
+        "swap_node_labels": 0.65,
+        "swap_subtrees": 0.25
+    },
+    "fnr": 0.1,
+    "fpr": 0.2,
+    "num_samples": 1000,
+    "burn_in": 20,
+    "thinning": 100
+}
+
+"""
+
+import argparse
+import jax.random as random
+import json
+import jax.numpy as jnp
+import logging
+
+from pathlib import Path
+from datetime import datetime
+
+import pyggdrasil.tree_inference as tree_inf
+import pyggdrasil.serialize as serialize
+
+from pyggdrasil.tree_inference import (
+    MutationMatrix,
+    mcmc_sampler,
+    MoveProbabilities,  # type: ignore
+)
+
+
+def create_parser() -> argparse.Namespace:
+    """
+    Parser for required input user.
+
+    Returns:
+        args: argparse.Namespace
+    """
+
+    parser = argparse.ArgumentParser(description="Run MCMC sampler for tree inference.")
+    parser.add_argument(
+        "--seed",
+        required=False,
+        help="Seed for random chain ( and tree generation if no tree is provided).",
+        type=int,
+        default=42,
+    )
+    parser.add_argument(
+        "--init_tree_fp",
+        required=False,
+        help="Fullpath of initial tree to start the MCMC sampler from.",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--init_tree_mcmc_no",
+        required=False,
+        help="Sample number if mcmc sample is provided as initial tree. "
+        "- line number in file from 1",
+        default=None,
+        type=int,
+    )
+
+    parser.add_argument(
+        "--init_TreeNode",
+        required=False,
+        help="Sample number if mcmc sample is provided as initial tree.",
+        default=None,
+        action="store_true",
+    )
+
+    parser.add_argument("--config_fp", required=True, help="Config file path", type=str)
+
+    parser.add_argument(
+        "--out_dir",
+        required=True,
+        help="Output directory to save the MCMC samples.",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--data_fp", required=True, help="File path containing the data.", type=str
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_mutation_matrix(data_fp: str) -> MutationMatrix:
+    """Load the mutation matrix from file.
+
+    Args:
+        data_fp: str
+            Filepath of the data.
+
+    Returns:
+        mut_mat: MutationMatrix
+            observed mutation matrix
+    """
+
+    # load data from file to json object
+    with open(data_fp, "r") as f:
+        data = json.load(f)
+
+    # convert json object to mutation matrix
+    mut_mat = data["noisy_mutation_mat"]  # TODO: adjust to flexible title
+    # convert to array
+    mut_mat = jnp.array(mut_mat)
+
+    return mut_mat
+
+
+def run_chain(params: argparse.Namespace, config: dict, **kwargs) -> None:
```

Comment:
For `config` it may be beneficial in the long run to make it a Pydantic object (my favourite solution), a data class, or a [TypedDict](https://peps.python.org/pep-0589/).